internal/fs/inode/core.go

```diff
@@ -48,6 +51,8 @@ func (c *Core) Type() metadata.Type {
 	switch {
 	case c == nil:
 		return metadata.UnknownType
+	case c.Folder != nil:
+		return metadata.ExplicitDirType
 	case c.MinObject == nil && !c.Local:
 		return metadata.ImplicitDirType
 	case c.FullName.IsDir():
```

Comment:
The first check performed is:
```
case c.MinObject == nil && !c.Local:
    return metadata.ImplicitDirType
```
This condition will always be evaluated first so folder will return as implicit_dir as c.MinObject will be nil. 

If the order of checks is changed, the existing flow will break because we currently prioritize checking for implicit directories before explicit ones.

Reply:
Added && `c.Folder == nil` condition in Implicit dir.